### PR TITLE
Chore: Correctly escape strings in JSONFormatter

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Fixes a "Incomplete string escaping or encoding" warning from code scanning. The warning itself is pretty benign as this function is only used for display formatting, and only in the Table panel I think?

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/security/code-scanning/9?query=ref%3Arefs%2Fheads%2Fmain

Relates to #43080

Before: 
image

After:
image

The new value is correct because when i copy-paste it into the JSON console, the string is logged as expected 

image

### CODE REVIEW

**Review by ashharrison90 — APPROVED** [review]
looks great! do we know why it was replacing `"` with `"` in the first place? seems odd 😕

**Comment by joshhunt** [comment]
The code was originally `str.replace('"', '\"')`, aiming to put a slash in-front of the double-quote in the source string, but this wouldn't actually work because the slash itself would needed to have been escaped.

Then the source was prettiered and it removed the redundant slash, which is how we ended up with the weird code now 😅

**Comment by daniellee** [comment]
See that you are struggling with the same warnings that I got after doing the first simple fix: https://github.com/grafana/grafana/pull/43529

**Comment by daniellee** [comment]
I think @leventebalogh 's point about having one place to sanitize strings is relevant though: https://github.com/grafana/grafana/pull/43529#issuecomment-1014234259

**Comment by joshhunt** [comment]
Yeah, I just saw that comment (and didn't realise you already had a PR open to tackle this!!) @daniellee. I think the intention might be a little bit different - the purpose of this function is not to sanitise or to help with security, but just for formatting and display. Having these clear and separate intentions might be a reason not to do that. Maybe.

**Review by mckn — APPROVED** [review]
Great work! Awesome with the added tests.

**Comment by daniellee** [comment]
@joshhunt you did see that you still have a CodeQL error as your regex doesn't escape slashes?

```
This does not escape backslash characters in the input.
```

**Comment by joshhunt** [comment]
Yes I did. haven't had the chance to pick this back up again 👍🏻

File: packages/grafana-ui/src/components/JSONFormatter/json_explorer/helpers.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@
 // License MIT, Copyright (c) 2015 Mohsen Azimi
 
 /*
- * Escapes `"` characters from string
+ * Escapes `"` characters from string for formatting
```
Comment (leventebalogh) [inline packages/grafana-ui/src/components/JSONFormatter/json_explorer/helpers.ts:5]: **Nit:** I don't think this comment adds to much extra information

I would probably rather change the name of the function in this case to something like `escapeStringForFormatting()` (as it's more likely that a comment will remain unchanged when the logic of the function is extended than the function name itself), and also either get rid of the comment, or explain the _why_. 

I think what's unclear for anyone new to this piece of code is _why_ we need to do this.

**Comment by leventebalogh** [comment]
> Yeah, I just saw that comment (and didn't realise you already had a PR open to tackle this!!) @daniellee. I think the intention might be a little bit different - the purpose of this function is not to sanitise or to help with security, but just for formatting and display. Having these clear and separate intentions might be a reason not to do that. Maybe.

I see, I think you are right in case it's just a local utility function that helps the visual formatting and has nothing to do with actual sanitisation. Maybe I would just make the name a bit less generic in that case.

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**